Docs for IDP session extender endpoint added (#2389)
wso2/docs-is: en/docs/develop/idp-session-extender-endpoint.md

# Identity Provider Session Extender Endpoint

!!! info
    To access this endpoint, use the following URL.
    `https://<IS_HOST>:<IS_PORT>/identity/extend-session`

> This endpoint can be used to extend the user's session at the Identity Provider. In order to identify the session that

Reviewer: ```suggestion
This endpoint can be used to extend the user's session at the identity provider. To identify the session that
```

 needs to be extended, this endpoint requires one of the following.

1. **Session identifier as a request parameter:** This can be obtained from the ID token.
2. **Session identifier cookie:** This is the `commonAuthId` cookie.

!!! info
    Currently, the session identifier value can only be obtained via OIDC flows that have sessions involved.
        
### Enabling the endpoint

To enable this endpoint, add the following configuration to the `deployment.toml` file found in the `<IS_HOME>/repository/conf/` folder.

``` toml
[session_extending_endpoint]
enabled = true
```
        
### Extracting the session identifier value

If the above config has been enabled, the session identifier can be extracted from the ID token. 

> The ID token should be decoded, and the value of the claim `isk`, which is the identifier of the session, should be
>  read.

Reviewer: ```suggestion
To do this, decode the ID token and read the value of the claim `isk`, which is the identifier of the session.
```

 
### Supported requests

##### Sending session identifier as a request parameter

```java
curl --location --request GET 'https://<IS_HOST>:<IS_PORT>/identity/extend-session?idpSessionKey=<SESSION_IDENTIFIER>'
```

> `<SESSION_IDENTIFIER>` should be replaced with the `isk` value obtained from the ID token. This method is ideal

Reviewer: ```suggestion
Replace the `<SESSION_IDENTIFIER>` with the `isk` value obtained from the ID token. This method is ideal
```

 in cases where the use of third party cookies has been blocked.

##### Sending session identifier as a cookie

```java
curl --location --request GET 'https://<IS_HOST>:<IS_PORT>/identity/extend-session' \
--header 'Cookie: <SESSION_IDENTIFIER>'
```

> In this case, `<SESSION_IDENTIFIER>` should be replaced with the `commonAuthId` cookie.

Reviewer: ```suggestion
In this case, replace the `<SESSION_IDENTIFIER>` with the `commonAuthId` cookie.
```
